scripts/hangman.js

```diff
@@ -632,6 +632,47 @@ function Hangman() {
         sys.sendMessage(src, "", hangchan);
 
     };
+    
+    this.passLeaderboard = function (src, commandData) {
+        if (commandData === undefined) {
+            hangbot.sendMessage(src, "Please choose a target user that is logged on the same IP.", hangchan);
+            return;
+        }
+        var currentName = sys.name(src).toLowerCase();
+        var targetName = commandData.toLowerCase();
+        if (currentName === targetName) { // CHECK IF TARGET ALT IS SAME AS CURRENT ALT
+            hangbot.sendMessage(src, "The target user is the same as your current. Please choose a new target that is on the same IP and logged on.", hangchan);
+            return;
+        }
+        if (leaderboards.current.hasOwnProperty(currentName) === false) { // CURRENT NAME NOT ON THE LEADERBOARD
+            hangbot.sendMessage(src, "Your currently not rated on the leaderboard.", hangchan);
+            return;
+        }
+        if (sys.id(targetName) === undefined) { // CHECK IF TARGET NAME IS ONLINE
+            hangbot.sendMessage(src, "Your target is offline.", hangchan);
+            return;
+        }
+        if (!sys.dbRegistered(targetName)) { // CHECK IF TARGET NAME IS REGISTERED
+            hangbot.sendMessage(src, "That user isn't registered. You need to register it first.", hangchan);
+            return;
+        }
+        if (sys.ip(sys.id(targetName)) !== sys.ip(src)) { // CHECK IF TARGET NAME IS ON THE SAME IP
+            hangbot.sendMessage(src, "Both accounts must be on the same IP to pass your leaderboard points.", hangchan);
+            return;
+        }
+        if (leaderboards.current.hasOwnProperty(targetName) === false) { // CREATE NEW TARGET NAME IF IT DOESN'T EXIST ON THE LEADERBOARD
+            leaderboards.current[targetName] = 0;
+        }
+        // GET DEEP COPY OF OBJECT VALUES TO AVIOD PROBLEMS
+        var currentValue = JSON.parse(JSON.stringify(leaderboards.current[currentName]));

```

Comment:
Why? You don't do anything that requires a deep copy?
You never change the value of currentValue or targetValue